controllers/topics/index.js

```diff
@@ -285,6 +285,25 @@ const getFollowerList = async (req, res) => {
   try {
     const searchQuery = search ? `AND LOWER(users.username) LIKE LOWER(:search)` : '';
 
+    let user_topics = await sequelize.query(
+      `select tp.topic_id from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = :userId`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        }
+      }
+    );
+    let topicIds = user_topics.map((topic) => topic.topic_id);
+    const similarTopicQuery =
+      topicIds.length > 0
+        ? `ARRAY( select name from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = "users".user_id and tp.topic_id in (:topicIds) limit 3)`
+        : 'ARRAY[]::text[]';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query used to fetch user topics and similar topics is vulnerable to SQL injection attacks. It's recommended to use parameterized queries or prepared statements to prevent this. Also, the `userId` in the WHERE clause should be parameterized.

```diff
- where utp.user_id = :userId`,
+ where utp.user_id = ?`,
```

324:
The `similarTopicQuery` is directly interpolated into the SQL query which can lead to SQL injection if not properly sanitized. Consider using a parameterized query instead.

357:
The `topicIds` variable is passed as a replacement for the SQL query but it's an array. This could potentially cause issues if the SQL driver doesn't support array replacements. You might need to convert the array to a string of comma-separated values before passing it to the query.

```diff
-         topicIds,
+         topicIds: topicIds.join(','),
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->